### PR TITLE
#2075 validation fixed for combination of disabled+another failed validation

### LINKED ISSUE #2075: API v3 - Put /{name}/{version}/properties returns bad validation
## Describe the bug
Put on dataset properties returns a slightly misleading error when the dataset is disabled and user tries to update properties, omitting the mandatory ones.

## To Reproduce
Steps to reproduce the behaviour OR commands run:
1. Have a mandatory property `PropertyA`
2. Have a dataset `DatasetX` that is `disabled = true`
3. run Put `/DatasetX/latest/properties` with empty body (or omitting `propertyA`)
4. Validation response, error part show only missing mandatory property but not the fact that `DatasetX` is disabled.

Response:
```json
{
  "errors": {
    "PropertyA": [
      "Dataset property 'PropertyA' is mandatory, but does not exist!"
    ]
  },
  "warnings": {}
}
```

## Expected behaviour

I would expect one of 2 results. Either I get a full list of errors, or I prefer the fact that the dataset is disabled. 

#### Full Validation response:
```json
{
  "errors": {
    "PropertyA": [
      "Dataset property 'PropertyA' is mandatory, but does not exist!"
    ],
    "disabled": [
      "Entity DatasetX is disabled!"
    ]
  },
  "warnings": {}
}
```

#### Just disabled
```json
{
  "errors": {
    "disabled": [
      "Entity DatasetX is disabled!"
    ]
  },
  "warnings": {}
}
```

### PR DESCRIPTION
**Pull request by dk1844**
This PR fixes a validation bug found in #2075.

In short, attempting to update properties to with an invalid payload (_Problem 1_) for a disabled dataset (_Problem 2_) behaved incorrectly, because only the invalid payload issue (Problem 1) was reported. (Problem 2 was only reported after Problem 1 has been fixed, which was considered too late).

This PR fixes the issue by removing an unnecessary validation of the properties update (that raised an exception without merging with other validation results). This was completely removed because the basic `DatasetServiceV3.validate()` already performs all the validation steps (including properties validation) and also correctly merges all validation outcomes to be reported to back to the user.

**A [test case ](https://github.com/AbsaOSS/enceladus/pull/2077/files#diff-968796a28b59cae88f5bf58aa45e1fec0899728819f99eee92cf97356e6fbb98R753)documenting this very behavior has been added**, it literally contains:
```scala
response1.getBody shouldBe Validation(Map(
  "disabled" -> List("Entity datasetA is disabled!"), // this is the most important validation to appear
  "unknown1" -> List("There is no property definition for key 'unknown1'."),
  "mandatoryA" -> List("Dataset property 'mandatoryA' is mandatory, but does not exist!")
))
```

 - unnecessary `VersionedModelServiceV3.update` was also removed, becase it added nothing over the ancestor functionality

Fixes #2075

### CODE REVIEW

**Review by Zejnilovic — APPROVED** [review]
I built it and run it

**Review by Zejnilovic — APPROVED** [review]
Code reviewed

File: rest-api/src/test/scala/za/co/absa/enceladus/rest_api/integration/controllers/DatasetApiIntegrationSuite.scala [hunk 1]
```diff
@@ -361,7 +361,7 @@ class DatasetApiIntegrationSuite extends BaseRestApiTestV2 with BeforeAndAfterAl
 
           assert(headers2.getFirst("Location").contains("/api/dataset/dataset/2"))
           assert(body2.version == 2)
-          assert(body2.properties == expectedPropertiesSet)
+          //TODO DatasetApiIntegrationSuite failing test on merge to Enceladus 3 #1949
```
Comment (AdrianOlosutean) [inline rest-api/src/test/scala/za/co/absa/enceladus/rest_api/integration/controllers/DatasetApiIntegrationSuite.scala:364]: No need to include this change, it seems like it works now

Comment (dk1844) [reply rest-api/src/test/scala/za/co/absa/enceladus/rest_api/integration/controllers/DatasetApiIntegrationSuite.scala]: Fixed.

**Review by AdrianOlosutean — DISMISSED** [review]
Code reviewed

File: README.md [hunk 2]
```diff
@@ -356,6 +356,8 @@ A module containing [examples](examples/README.md) of the project usage.
 ## How to contribute
 Please see our [**Contribution Guidelines**](CONTRIBUTING.md).
 
+## Documentation
```
[the remark this answers is not in the document]
Comment (dk1844) [reply README.md:359]: Thanks, while merging, I have made a wrong decision, apparently, not seeing the bigger picture. Fixed.

File: spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala [hunk 3]
```diff
@@ -19,12 +19,9 @@ import java.text.MessageFormat
 import java.time.Instant
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.FileSystem
-import org.apache.spark.SPARK_VERSION
 import org.apache.spark.sql.functions.{lit, to_date}
 import org.apache.spark.sql.{DataFrame, SparkSession}
 import org.apache.spark.{SPARK_VERSION, SparkConf}
```
[the remark this answers is not in the document]
Comment (dk1844) [reply spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala]: Thanks for noticing, removed.

**Review by benedeki — APPROVED** [review]

**Pull request merged**